### PR TITLE
Fix Typescript 4

### PR DESCRIPTION
**Pull request by RyanMacBern**
# Description

Previously the project used Typescript v4, but the build used Typescript v3. This PR swaps `tsdx` (abandoned before Typescript v4) with `dts-cli` (forked from tsdx to add Typescript v4 support).

This change allows us to use Typescripts's template literals for [the Drawer PR](https://github.com/palmetto/palmetto-components/pull/752).

This change also resolves issues when testing locally in a consuming project. Documentation has been updated to use local package dependency rather than symlinking.

# What type of change is this?

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Updating documentation.
- [x] Updating deployment/build pipeline.
- [x] Updating Project Dependencies

# Completeness Checklist

- [x] TESTS: My changes maintain the baseline required test coverage, as specified by code climate analysis.
- [ ] DOCS: All new component work is covered in that component's `Overview` docs.
- [ ] DOCS: All new component work is covered in a component `Playground`.
- [ ] DOCS: All new visual changes or options are covered under relevant components' `VisualTests`.
- [x] My code has no linting or typescript compile warnings.
- [ ] My work is tied to a Github issue and satisfies the acceptance criteria (if applicable) of the corresponding issue.

# Quality Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

# UI Checklist
- [ ] I have conducted visual UAT on my changes/features.
- [ ] My solution works well on desktop, tablet, and mobile browsers.

### CODE REVIEW

File: src/components/Toast/Toast.test.tsx [hunk 1]
```diff
@@ -259,6 +253,7 @@ describe('Toast', () => {
 
       act(() => {
         jest.advanceTimersByTime(1000);
+        jest.useRealTimers();
```
Comment (juanfabrega) [inline src/components/Toast/Toast.test.tsx:256]: @RyanMacBern curious why we need to manually set jest to useRealTimers if this runs after every test.

**Review by juanfabrega — APPROVED** [review]
@RyanMacBern thanks for the great work here! I am disappointed with how quickly TSDX fell off a cliff, but it seems at least the fork is currently active and going strong.

File: src/components/Toast/Toast.test.tsx (hunk 1 above)
Comment (RyanMacBern) [reply src/components/Toast/Toast.test.tsx:256]: This test started failing after the upgrade. I spent quite a bit of time messing with it, and this is the only thing that worked. For whatever reason, it needs to switch to real timers **before** the last expect or the text will not be in the document.

**Review by nathanyoung — APPROVED** [review]
🎉

**Pull request merged**